tedana/selection/select_comps.py

```diff
@@ -210,50 +383,98 @@ def selcomps(seldict, mmix, mask, ref_img, manacc, n_echos, t2s, s0, olevel=2,
     epsmap = []
     Rhos_sorted = np.array(sorted(seldict['Rhos']))[::-1]
     # Make an initial guess as to number of good components based on
     # consensus of control points across Rhos and Kappas
+    # For terminology later, typically getelbow _aggr > _mod > _cons
+    #  though this might not be universally true. A more "inclusive" threshold
+    #  has a lower kappa since that means more components are above that thresh
+    #  and are likely to be accepted. For Rho, a more "inclusive" threshold is
+    #  higher since that means fewer components will be rejected based on rho.
+    # KRcut seems weird to handwerkerd. I see that the thresholds are slightly
+    # shifted for kappa & rho later in the code, but why would we ever want to
+    # set a common threhsold reference point for both? These are two different
+    # elbows on two different data sets.
     KRcutguesses = [getelbow_mod(seldict['Rhos']), getelbow_cons(seldict['Rhos']),
                     getelbow_aggr(seldict['Rhos']), getelbow_mod(seldict['Kappas']),
                     getelbow_cons(seldict['Kappas']), getelbow_aggr(seldict['Kappas'])]
+    KRcut = np.median(KRcutguesses)
+
+    # Also a bit weird to handwerkerd. This is the 75th percentile of Kappa F
+    # stats of the components with the 3 elbow selection criteria and the
+    # F states for 3 significance thresholds based on the # of echos.
+    # This is some type of way to get a significance criterion for a component
+    # fit, but it's include why this specific criterion is useful.
     Khighelbowval = stats.scoreatpercentile([getelbow_mod(seldict['Kappas'], val=True),
                                              getelbow_cons(seldict['Kappas'], val=True),
                                              getelbow_aggr(seldict['Kappas'], val=True)] +
                                             list(utils.getfbounds(n_echos)),
                                             75, interpolation_method='lower')
-    KRcut = np.median(KRcutguesses)
 
-    # only use exclusive when inclusive is extremely inclusive - double KRcut
+    # Default to the most inclusive kappa threshold (_cons) unless:
+    # 1. That threshold is more than twice the median of Kappa & Rho thresholds
+    # 2. and the moderate elbow is more inclusive than a p=0.01
+    # handwerkerd: This actually seems like a way to avoid using the theoretically
+    #   most liberal threshold only when there was a bad estimate and _mod is
+    #   is more inclusive. My one concern is that it's an odd way to test that
+    #   the _mod elbow is any better. WHy not at least see if _mod < _cons?
     cond1 = getelbow_cons(seldict['Kappas']) > KRcut * 2
     cond2 = getelbow_mod(seldict['Kappas'], val=True) < F01
     if cond1 and cond2:
         Kcut = getelbow_mod(seldict['Kappas'], val=True)
     else:
         Kcut = getelbow_cons(seldict['Kappas'], val=True)
-    # only use inclusive when exclusive is extremely exclusive - half KRcut
```

Comment:
Is there a reason to remove this comment ? It might be helpful to have the original explanation, even if we decide that it doesn't make sense (for us).

Reply:
I tried to expand the purpose of that comment since it's not clear how inclusive & exclusive are defined. I felt like the following blocks of comments added more detail and replaced this one. If you think the line I deleted contains unique and relevant info, I can add it back in.
https://github.com/handwerkerd/tedana/blob/4c65baf8873ebbc4965820101b006c2cc061c4b7/tedana/selection/select_comps.py#L399-L409
https://github.com/handwerkerd/tedana/blob/4c65baf8873ebbc4965820101b006c2cc061c4b7/tedana/selection/select_comps.py#L415-L419
https://github.com/handwerkerd/tedana/blob/4c65baf8873ebbc4965820101b006c2cc061c4b7/tedana/selection/select_comps.py#L426-L432

Reply:
What about keeping it with something like "Originally written by PK:" at the start of the comment ? That way, if in finishing commenting this code we decide that it adds something helpful, we can work it back in -- and otherwise delete it. Does that seem reasonable, or do you think it's confusing to have even now ?

Reply:
OK. I also removed some comments on Step numbers since it was completely unclear to me why a bunch of stuff was or wasn't considered part of each step. I'll add those back in as well.

Reply:
That seems reasonable to me, at least for the time being (where we're denoting them as "originally written by" or whatever you think best). Thanks again, @handwerkerd !